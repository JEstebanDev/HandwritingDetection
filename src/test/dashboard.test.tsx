import { render, screen } from "@testing-library/react";
import { MemoryRouter } from "react-router-dom";
import Dashboard from "../src/dashboard";

describe("Testing the dashboard components", () => {
  test("should render the component Dashboard", () => {
    const { container } = render(
      <MemoryRouter initialEntries={["/"]}>
        <Dashboard />
      </MemoryRouter>
    );
    expect(container).toMatchSnapshot();
  });

  test("should render the text", () => {
    const { container } = render(
      <MemoryRouter initialEntries={["/"]}>
        <Dashboard />
      </MemoryRouter>
    );
    expect(screen.getByText(/hey, drop me some files/i));
  });
});
