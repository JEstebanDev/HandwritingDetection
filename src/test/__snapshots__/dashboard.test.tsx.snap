// Vitest Snapshot v1

exports[`Testing the dashboard components > should render the component Dashboard 1`] = `
<div>
  <div
    class="container"
  >
    <div
      class="FilesDragAndDrop"
    >
      <div
        class="FilesDragAndDrop__area"
      >
        Hey, drop me some files
        <span
          aria-label="emoji"
          class="area__icon"
          role="img"
        >
          ðŸ˜Ž
        </span>
      </div>
    </div>
    <div
      class="listCards"
    />
  </div>
</div>
`;

exports[`first > should first 1`] = `
<div>
  <div
    class="container"
  >
    <div
      class="FilesDragAndDrop"
    >
      <div
        class="FilesDragAndDrop__area"
      >
        Hey, drop me some files
        <span
          aria-label="emoji"
          class="area__icon"
          role="img"
        >
          ðŸ˜Ž
        </span>
      </div>
    </div>
    <div
      class="listCards"
    />
  </div>
</div>
`;

exports[`first > should first 2`] = `
<div>
  <div
    class="container"
  >
    <div
      class="FilesDragAndDrop"
    >
      <div
        class="FilesDragAndDrop__area"
      >
        Hey, drop me some files
        <span
          aria-label="emoji"
          class="area__icon"
          role="img"
        >
          ðŸ˜Ž
        </span>
      </div>
    </div>
    <div
      class="listCards"
    />
  </div>
</div>
`;

exports[`first > should render the component Dashboard 1`] = `
<div>
  <div
    class="container"
  >
    <div
      class="FilesDragAndDrop"
    >
      <div
        class="FilesDragAndDrop__area"
      >
        Hey, drop me some files
        <span
          aria-label="emoji"
          class="area__icon"
          role="img"
        >
          ðŸ˜Ž
        </span>
      </div>
    </div>
    <div
      class="listCards"
    />
  </div>
</div>
`;
